key.js

```diff
@@ -1,106 +1,350 @@
+        ];
+        sounds.forEach(audio => audio.pause());
+    }
+    function unMuteAudio() {
+        gameState.sound = true;
+        const sounds = [
+            gameState.gameAudio.getGameAudio(),
+            gameState.gameAudio.getStartSound(),
+            gameState.gameAudio.getClickAudio(),
+            gameState.gameAudio.getKeyboardSounds.getInvalidAudio(),
+            gameState.gameAudio.getKeyboardSounds.getGreenAudio(),
+            gameState.gameAudio.getKeyboardSounds.getRedAudio()
+        ];
+        sounds.forEach(audio => {
+            audio.currentTime = 0;
+            audio.play()
+        });
+    }
+    function decrementGameLife() {
+        gameState.lives--;
+    }
+    function printGameState() {
+        console.log(gameState);
     }
-});
 
-const h3 = document.querySelector('h3');
+    return {
+        startGame,
+        gameOver,
+        levelUp,
+        resetGameLives,
+        resetGameTimer,
+        resetGameState,
+        muteAudio,
+        unMuteAudio,
+        decrementGameLife,
+        printGameState,
+        getGameLife,
+        endGame,
+        gameMusicPlay,
+        gameMusicStop
+    };
+    // I. create a game state object (ensure it's a deep copy of initialGameState)
+    // II. create utility methods to manipulate the game state object
+    // 1. incrementGameLevel
+    // 2. getGameLevel
+    // 3. startGame
+    // 4. endGame
+    // 5. isGameRunning
+    // 6. isGameOver
+    // 7. getGameLife
+    // 8. decrementGameLife
+    // 9. resetGameState
+    // 10. getGameAudio
+    // 11. printGameState // just for debugging
+    // 12. <create any other method you feel that makes sense>
+    // III. return an object with all the aforementioned utility methods from 1..9.
 
-// Random KeyGenerator :
-const span = document.querySelectorAll('span');
-const arr = [];
-const newSequence = () => {
-    let i = 10;
-    while (i--) {
-        const random = Math.random();
-        if (random <= .25) {
-            arr.push("&uarr;");
-        }
-        else if (random <= .50) {
-            arr.push("&rarr;");
-        }
-        else if (random <= .75) {
-            arr.push("&rarr;");
+    // return {
+    //     incrementGameLevel: () => gameState.levelsInfo.currentLevel++
+    // }
+})();
+// IV. Refactor the code below to use the gameModule above for creating the game logic.
+// Button Animation Logic :
+
+const viewModule = (function () {
+    // Global Timer Variable :
+
+    let timer;
+    function StartTimer() {
+        timer = setInterval(() => {
+            gameState.seconds = gameState.seconds - 1;
+            timerElement.textContent = `${gameState.seconds}`;
+            if (gameState.seconds <= 0) {
+                clearInterval(timer);
+                const secondsReset = initialGameState.seconds;
+                gameState.seconds = secondsReset;
+            }
+        }, 1000)
+    }
+    function StopTimer() {
+        clearInterval(timer);
+    }
```

Comment:
Add a `null` or `undefined` check, without that, you can get an error when you're trying to clear a `null` or `undefined` `oneSecondTimer` variable.

Also, the function name has to follow camel casing.  Please use a ES5 linter on VS code.

```suggestion
    function stopTimer() {
        if (!timer) return;
        clearInterval(timer);
    }
```